src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+    private K key;
+    private V value;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == null && key == null)
+                    || ((keys[i] == key) || (keys[i] != null && keys[i].equals(key)))) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == null && key == null)
+                    || ((keys[i] == key) || (keys[i] != null && keys[i].equals(key)))) {
+                return values[i];
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        int size = 0;
+        for (int i = 0; i < values.length; i++) {
+            if (values[i] != null) {
+                size++;
+            }
+        }
+        return size;
+    }
+
+    private V info(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == null && key == null)
+                    || ((keys[i] == key) || (keys[i] != null && keys[i].equals(key)))) {
+                return values[i];
+            }
+        }
+        return null;
     }
```

Comment:
```suggestion
    private V getValueByKey(K key) {
        for (int i = 0; i < size; i++) {
            if ((keys[i] == null && key == null)
                    || ((keys[i] == key) || (keys[i] != null && keys[i].equals(key)))) {
                return values[i];
            }
        }
        return null;
    }
```
You don't use this method anywhere think about what code in your methods you could replace with this method.